doc/source/algorithms.rst

```diff
@@ -184,12 +184,33 @@ The user can (optionally) provide a visitor that perform operations along with t
 
     Invoked when a vertex is closed (all its neighbors have been examined).
 
-.. py:function:: dijkstra_shortest_paths_explicit(graph, edge_dists, source[, visitor])
-Returns the explicit paths using ``dijkstra_shortest_paths()`` as an array of
-vectors of vertices from the source to each target. Empty vectors are used to
-indicate vertices that are unreachable from the source. Parameters are the same
-as described in ``dijkstra_shortest_paths()``.
+.. py:function:: enumerate_paths(result[, dest])
 
+    Returns an array of vectors ``all_paths`` (containing vertices), where ``all_paths[i]`` corresponds to the path from a source to vertex ``dest[i]``. Empty vectors are used to indicate vertices that are unreachable from the source. ``dest`` can be a subset of vertices, or left unspecified (in which case, all the vertices in the graph will be considered). If ``dest`` is a single vertex, then ``all_paths`` is just an array of vertices, corresponding to the path from a source to ``dest``.
```

Comment:
This looks much better. Just a style point, but it's not necessary to name the output vector ``all_paths`` (whose name doesn't quite make sense anymore). I would just refer to the output vector without a name.